card/mod/core/set/all/fetch_helper.rb

```diff
@@ -72,13 +72,16 @@ def retrievable_mark_type_and_value mark
     end
   end
 
-  def retrieve_from_db? card, opts
-    card.nil? || (opts[:look_in_trash] && card.new_card? && !card.trash)
+  def retrieve_from_db? card, mark_type, mark_key, look_in_trash
+    return false if card&.real?                                      # real card found in cache
+    return false if mark_type === :key && mark_key.to_name.relative? # no relative names in db
+    return true if card.nil?                                         # card not found in cache
+    look_in_trash && !card.trash
```

Comment:
I should flag that this is another change in fetch logic.  I was seeing lots of lookups in the logs for things like "+topic", which we know is not going to be in the database, so I added the line about "no relative names".

Reply:
We don't allow `+topic` but it's possible to create cards like `_main`, `_user`, `_left` and `relative?` is true for those,

Reply:
should that be allowed?

Really the problem for me was that it kept looking in the db for the relative cards load after load.  So they weren't getting in the cache AND they kept making db calls.

Reply:
Probably better not to allow it. Even if we make them fetchable again they will have other issue (in nests for example). 

Do you know where these fetch calls for relative cards come from? Is that also bad code or justified for some reason?

Reply:
I'm not sure where it comes from, but my guess is that it's broadly used in places where we don't know what kind of name we're going to get.  Eg, we might look up a nested card with something like `Card.fetch nestedcardname, new: {}`, and in the case of a new form, all the field names will look like this.

I think it's fine to support fetching relative cards and changing their name later.  You can still do `Card.fetch("+topic", new: {})` and get a response, but we won't bother looking in the database for it.  (I wrote it to get silly queries out of the log).

Fwiw, I don't think this changes any behaviors; it's just an optimization. I only wanted to flag it because the last time I changed things around in fetch_helpers, I accidentally broke stuff.  

But are we agreeing that we don't want to allow cards with relative names to get stored in the database?  Basically, we just need to validate that a name is not relative when we do all our other name validations, right?